src/search_algo.hpp

```diff
@@ -568,25 +629,24 @@ search(LocalDataHolder<TGlobalHolder> & lH)
                         desiredOccs = 1;
 
                     // This aborts when we fall under the threshold
-                    auto old_cursor = cursor;
                     size_t old_count = cursor.count();
                     while (seedBegin + seedLength < lH.redQrySeqs[i].size())
                     {
-                        if (!cursor.extend_right(lH.redQrySeqs[i][seedBegin + seedLength]))
+                        auto newCursor = cursor.extendRight((lH.redQrySeqs[i][seedBegin + seedLength]).to_rank()+1);
+                        if (newCursor.empty()) {
                             break;
+                        }
 
-                        size_t new_count = cursor.count();
+                        size_t new_count = newCursor.count();
 
                         if (new_count < desiredOccs && new_count < old_count) // we always continue to extend if we don't loose anything
                         {
                             // revert last extension
-                            cursor = old_cursor;
                             break;
                         }
-
                         ++seedLength;
+                        cursor = newCursor;
                         old_count = new_count;
-                        old_cursor = cursor;
```

Comment:
Are the changes in this section semantic or did you just switch "old <-> current" to "current <-> new" ? It is a little bit confusing, because there is also old_count. 

I don't think it was ideal before, but I am not sure it is better now 😕

Reply:
I will revert this to be closer to the original.